zebra/zebra_nb_state.c

```diff
@@ -22,6 +22,21 @@
 #include "libfrr.h"
 #include "zebra_nb.h"
 #include "zebra/interface.h"
+#include "zebra/zebra_router.h"
+#include "zebra/debug.h"
+
+static void zebra_extract_afi_safi_key(char *key, int len, afi_t afi,
+				       safi_t safi, const char *mname)
+{
+	if (afi == AFI_IP && safi == SAFI_UNICAST)
+		snprintf(key, len, "%s:%s", mname, "ipv4-unicast");
+	if (afi == AFI_IP6 && safi == SAFI_UNICAST)
+		snprintf(key, len, "%s:%s", mname, "ipv6-unicast");
+	if (afi == AFI_IP && safi == SAFI_MULTICAST)
+		snprintf(key, len, "%s:%s", "frr-zebra", "ipv4-multicast");
+	if (afi == AFI_IP6 && safi == SAFI_MULTICAST)
+		snprintf(key, len, "%s:%s", "frr-zebra", "ipv6-multicast");
```

Comment:
I think it would be better to return `const char *` here and use `snprintf()` in the call sites.